src/common/transformations/src/transformations/common_optimizations/concat_to_broadcast.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/concat_to_broadcast.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/tile.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+static bool use_broadcast(const std::shared_ptr<ov::op::v0::Concat>& concat) {
+    const auto& output = concat->output(0);
+    const auto& input = concat->input(0);
+    const auto& input_concat_dim = input.get_partial_shape()[concat->get_concatenation_axis()];
```

Comment:
should we check if the rank is static?

Reply:
already checked, see line 32 in the transformation predicate